homeassistant/components/wmspro/config_flow.py

```diff
@@ -38,7 +39,19 @@ async def async_step_dhcp(
         """Handle the DHCP discovery step."""
         unique_id = format_mac(discovery_info.macaddress)
         await self.async_set_unique_id(unique_id)
-        self._abort_if_unique_id_configured()
+
+        entry = self.hass.config_entries.async_entry_for_domain_unique_id(
+            DOMAIN, unique_id
+        )
+        if entry:
+            try:  # Check if current host is a valid IP address
+                ipaddress.ip_address(entry.data[CONF_HOST])
+            except ValueError:  # Do not touch name-based host
+                self._abort_if_unique_id_configured()
```

Comment:
in that case, we should abort always, regardless if the unique ID matches or not.

Reply:
In what case? It basically will always abort, because an entry with the same unique_id was found and checked for existence in the if statement going into this block. So the only difference here is if we want to update the CONF_HOST or not.

Do you suggest to replace the conditioned function call with the general call to abort with "already_configured" in the except case?

Reply:
Thanks and agreed. Just pushed the corresponding change.